### DISCUSSION TITLE
How does Printf work?

### DISCUSSION THREAD

**Discussion by CalebJohnHunt** [charmbracelet/bubbletea#367] [Q&A] [answered]
EDIT: I realized I was very wrong in how I thought this all worked. Nevertheless, I figured I'd just edit this discussion instead of making a new one.

How does one use the `Program.Println` method?

I thought perhaps you had to give a pointer to the Program to your model, but when I do (see below), I still get no output.

```go
package main

import (
	"log"

	tea "github.com/charmbracelet/bubbletea"
)

type model struct {
	p *tea.Program
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.Type.String() {
		case "ctrl+c":
			return m, tea.Quit
		case "p":
			m.p.Println("Why can't I see this?")
		}
	}
	return m, nil
}

func (m model) Init() tea.Cmd {
	return nil
}

func (m model) View() string {
	return "This is the view"
}

func main() {
	m := &model{}
	p := tea.NewProgram(m)
	m.p = p
	if err := p.Start(); err != nil {
		log.Fatal(err)
	}
}
```

**Comment by bashbunni** [comment, marked as answer]
Hey @Kaychub ! Thanks for reaching out.

A few things, the switch needs to be `switch msg.String()` to register the keypress and instead of `m.p.Println("Why can't I see this?")` it should be `tea.Println("Why can't I see this")`. I'll look into updating our docs on pkg.go.dev to include an example. I hope that helps clear things up! Let us know if you have any other questions

[Here](https://gist.github.com/bashbunni/7f87d5e4cb61b936a7469a9719707a80) is the working version of what you're aiming to accomplish

**Reply by CalebJohnHunt** [reply]
Oh wow. I don't know how I got that switch part so wrong. I swear I had it working before. Oh well.

As for the `tea.Println`, that totally makes sense! I was thinking it would immediately print something, but instead it gives a command which will then print something out. It seems so obvious now.

Thanks for the code example, and thanks for answering!

**Reply by bashbunni** [reply]
My pleasure! Please don't hesitate to reach out with questions in the future. It helps us a lot in finding gaps in our documentation
